lib/honeybadger/plugins/net_http.rb

```diff
@@ -0,0 +1,49 @@
+require 'net/http'
+require 'honeybadger/plugin'
+require 'honeybadger/instrumentation'
+require 'resolv'
+
+module Honeybadger
+  module Plugins
+    module Net
+      module HTTP
+        def request(request_data, body = nil, &block)
+          return super unless started?
+          return super if hb?
+
+          Honeybadger::Instrumentation.monotonic_timer { super }.tap do |duration, response_data|
+            context = {
+              duration: duration,
+              method: request_data.method,
+              status: response_data.code.to_i
+            }.merge(parsed_uri_data(request_data))
+
+            Honeybadger.event('request.net_http', context)
+          end[1] # return the response data only
+        end
+
+        def hb?
+          address.to_s[/#{Honeybadger.config[:'connection.host'].to_s}/]
+        end
+
+        def parsed_uri_data(request_data)
+          uri = request_data.uri || build_uri(request_data)
```

Comment:
Is there a concern with sensitive data in the URI here?

Reply:
That's a good question. There could be sensitive data here that a user may not know is being logged. Perhaps we need a mechanism to disable this by default.

Reply:
I added a default so this is disabled by default and must be explicitly enabled:
```
net_http:
  insights:
    enabled: true
```

Reply:
How about leaving it enabled but reporting just the host name?

Reply:
That works for me. I'll make it configurable to allow the full uri if set to true.